test/integration/fuzz/sanitizers/test.sh

```diff
@@ -0,0 +1,207 @@
+            journalctl -D "/var/log/journal/${CONTAINER_ID:?}" -b -o short-monotonic >"$TMT_TEST_DATA/container-journal.log"
+        fi
+
+        rm -rf "/var/lib/machines/$CONTAINER"
+        rm -rf "/var/log/journal/$CONTAINER_ID"
+        rm -rf "/run/systemd/system/systemd-nspawn@$CONTAINER.service.d"
+        systemctl daemon-reload
+    fi
+}
+
+trap at_exit EXIT
+
+# Switch SELinux to permissive (if enabled), so it doesn't interfere with the container shenanigans below.
+setenforce 0 || :
+# We need persistent journal for the systemd-nspawn --link= stuff
+mkdir -p /var/log/journal
+journalctl --flush
+
+: "=== Prepare dbus-broker's source tree ==="
+# Since we need to build dbus-broker from scratch, let's do some magic to get the correct source tree.
+if [[ -n "${PACKIT_TARGET_URL:-}" ]]; then
+    # If we're running in Packit's context, use the set of provided environment variables to checkout the
+    # correct branch (and possibly rebase it on top of the latest source base branch so we always test the
+    # latest revision possible).
+    git clone "$PACKIT_TARGET_URL" dbus-broker
+    cd dbus-broker
+    git checkout "$PACKIT_TARGET_BRANCH"
+    # If we're invoked from a pull request context, rebase on top of the latest source base branch.
+    if [[ -n "${PACKIT_SOURCE_URL:-}" ]]; then
+        git remote add pr "${PACKIT_SOURCE_URL:?}"
+        git fetch pr "${PACKIT_SOURCE_BRANCH:?}"
+        git merge "pr/$PACKIT_SOURCE_BRANCH"
+    fi
+    git log --oneline -5
+elif [[ -n "${DBUS_BROKER_TREE:-}" ]]; then
+    # Useful for quick local debugging when running this script directly, e.g. running
+    #
+    #   $ DBUS_BROKER_TREE=$PWD test/integration/fuzz/sanitizers/test.sh
+    #
+    # from the dbus-broker repo root.
+    cd "${DBUS_BROKER_TREE:?}"
+else
+    # If we're running outside of Packit's context, pull the latest dbus-broker upstream.
+    git clone https://github.com/bus1/dbus-broker dbus-broker
+    git log --oneline -5
+fi
+
+
+: "=== Build dbus-broker with sanitizers and run the unit test suite ==="
+meson setup build-san --wipe -Db_sanitize=address,undefined -Dprefix=/usr
+ninja -C build-san
+meson test -C build-san --timeout-multiplier=2 --print-errorlogs
+
+
+: "=== Run tests against dbus-broker running under sanitizers ==="
+# So, this one is a _bit_ convoluted. We want to run dbus-broker under sanitizers, but this bears a couple of
+# issues:
+#
+#   1) We need to restart dbus-broker (and hence the machine we're currently running on)
+#   2) If dbus-broker crashes due to ASan/UBSan error, the whole machine is hosed
+#
+# To make the test a bit more robust without too much effort, let's use systemd-nspawn to run an ephemeral
+# container on top of the current rootfs. To get the "sanitized" dbus-broker into that container, we need to
+# prepare a special rootfs with just the sanitized dbus-broker (and a couple of other things) which we then
+# simply overlay on top of the ephemeral rootfs in the container.
```

Comment:
I can't say I fully understand what's going on here :-) but assuming the next step would be to run it all under Valgrind too I wonder if it's possible to somehow split the script and move the test suite itself to a separate bash script that could be run in two different places?

Reply:
It _should_ be relatively easy to factor the container shenanigans out into a separate helper, I'll check what's the actual reality :)

Reply:
I have one more question. Is it run on VMs/bare-metal machines where coredumps can be collected? I'm not sure what should happen when dbus-broker crashes/segfaults or whatever.

Reply:
Ah, good point. Until recently systemd-coredump would leave coredumps from containers on the host, but since systemd/systemd@a108c43e36d3ceb6e34efe37c014fc2cda856000 it forwads the coredumps to the container to handle, so this might need some extra attention as well. We could either disable the forwarding (`CoredumpReceive=no` in the respective `systemd-nspawn@.service` instance) and just use `coredumpctl` on the host, configure `systemd-journald` to store the coredumps in the journal (since the container journal is already exported on the host), or bind mount a directory from the host to `/var/lib/systemd/coredump` so the coredumps don't vanish together with the container.

I like the first option the most (and since we use host's /usr we shouldn't have any issues with mismatches symbols), but I'll play with this a bit to see how well it works in practice.

Reply:
(Apologies for dropping the ball on this, got very busy with some internal stuff. Hopefully I'll get back to this soon.)